junit-runtime/src/main/scala/org/scalajs/junit/JUnitTask.scala

```diff
@@ -86,8 +86,7 @@ private[junit] final class JUnitTask(val taskDef: TaskDef,
     } yield {
       errors match {
         case e :: Nil if isAssumptionViolation(e) =>
-          reporter.reportIgnored(None)
-          ignored += 1
+          reporter.reportAssumptionViolation(None, timeInSeconds, e)
```

Comment:
At this point, it might be worth consolidating `executeTests` / `executeTestMethod` more. I think the current design largely is because this distinction needed to be made.

Reply:
I consolidated the reporting of errors in a separate method now. However, the rest of `executeTests`/`executeTestMethod` still seems different enough not to consolidate them further.